### PR TITLE
changeset completeness check

### PR DESCRIPTION
**Pull request by edwardfoyle**
## Problem

Our current changeset checks assert that PR that touch public packages have a changeset, but it does not check that all packages with changes are actually included in a changeset. This puts an unnecessary manual burden on the PR author and reviewer to ensure that all modified packages are included in a changeset.

**Issue number, if available:**

## Changes

Add a PR check to validate that any modified package is included in a changeset

**Corresponding docs PR, if applicable:**

## Validation

Running some experiments on this PR

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [x] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by 0618 — DISMISSED** [review]

**Review by bombguy — APPROVED** [review]

File: scripts/check_changeset_completeness.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  if (!packagesWithChangeset.has(modifiedPackageName)) {
+    packagesMissingChangesets.push(modifiedPackageName);
+  }
+}
+
+if (packagesMissingChangesets.length > 0) {
+  throw new Error(
+    `The following packages have changes but are not included in any changeset:${EOL}${EOL}${packagesMissingChangesets.join(
+      EOL
+    )}${EOL}${EOL}Add a changeset using 'npx changeset add' or 'npx changeset add --empty' if this is a non-functional change.`
```
Comment (sobolk) [inline scripts/check_changeset_completeness.ts:47]: I think message now should clarify that `--empty` is allowed if only private packages are changed.

**Review by sobolk — APPROVED** [review]

File: scripts/check_changeset_completeness.ts (hunk 1 above)
Comment (edwardfoyle) [reply scripts/check_changeset_completeness.ts:47]: Things like adding tests or linting could also have an empty changeset. Although maybe those cases are small enough that it's not worth mentioning

Comment (edwardfoyle) [inline scripts/check_changeset_completeness.ts:47]:
```suggestion
    )}${EOL}${EOL}Add a changeset using 'npx changeset add'.`
```

Comment (sobolk) [reply scripts/check_changeset_completeness.ts]: Good point.
We could perhaps have some predicate function in the algorithm that narrows down this classification.

something like

```
const shouldHaveNonEmptyChangeSet: () : 'yes'|'no'|'maybe' => {

if (API.md changed) {
 return yes
}
if (Only Test Changed) {
return no
}

return maybe;
}
```

Comment (edwardfoyle) [reply scripts/check_changeset_completeness.ts]: I'll leave this as a future optimization if needed. For now, I just removed the empty changeset suggestion. Having an unnecessary version bump is much better than a missing version bump.

**Review by 0618 — APPROVED** [review]
